flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveTableSource.java

```diff
@@ -126,7 +127,7 @@ public boolean isBounded() {
 	@Override
 	public DataStream<BaseRow> getDataStream(StreamExecutionEnvironment execEnv) {
 		if (!initAllPartitions) {
-			initAllPartitions();
+			initAllPartitions(null);
```

Comment:
Can we just `getPartitionInfos` here?
We don't need `initAllPartitions` and `partitionSpec2HiveTablePartition` in the class member.

Reply:
Do you mean we should only compute `allHivePartitions` here and not `partitionSpec2HiveTablePartition`? It might be a valid improvement but doesn't seem to be directly related to the goal of this PR. Besides, if we somehow firstly call `HiveTableSource::getDataStream ` and then call `HiveTableSource::applyPartitionPruning`, we may end up not computing `partitionSpec2HiveTablePartition` at all. Let's figure this out in a follow-up.

Reply:
I prefer do this little refactor. Because we don't need `partitionSpec2HiveTablePartition`. There is a little confuse about `initAllPartitions`.
There is no reason to get partition details except in `getDataStream`.